### ISSUE TITLE
Support fetching data from private datasets

### ISSUE THREAD

**Issue by marks** [timwis/vizwit#141]
Many do not realize it, but Socrata hosts more than just public data. I think this tool would be great for dashboards of private data, if it can be done in a secure way.

I'm thinking...
- the data would need to be fetched server side
- the user prompted for a username and password each time the page loaded. 
- OAuth
- an embed on a same-domain page may work without modifications (need to test)

**Comment by marks** [comment]
The last idea, _an embed on a same-domain page may work without modifications_ does work but may have limited utility to others

**Comment by timwis** [comment]
Good idea. I've personally tried to use vizwit for a private dataset and it was a real pain! Ended up creating an nginx reverse proxy on my local computer to pass through the requests as authenticated ones.

I think this would be pretty doable using the [authentication gateway chris metcalf setup](https://dev.socrata.com/changelog/2015/10/27/private-api-docs.html). The only catch would be whether we could implement it without making vizwit too socrata-specific? ie. a way to do it just in the socrata provider, or make vizwit just support oauth? I Have pretty limited knowledge of oauth.

**Comment by marks** [comment]
Update: first pass is working with [this config](https://gist.github.com/marks/014398166410456ada6d6b98bf285ad5). As @timwis suggested, this can be achieved using @chrismetcalf's https://github.com/socrata/proxy.dev.socrata.com

TODO:
- [ ] test with more than just the table card
- [ ] wait for https://github.com/socrata/soda-js/pull/22 to be accepted and switch package.json back to `socrata/soda-js`
- [ ] once merged, document `sodaProxy`
